files/zh-cn/web/javascript/reference/global_objects/string/split/index.md

```diff
@@ -5,111 +5,143 @@ slug: Web/JavaScript/Reference/Global_Objects/String/split
 
 {{JSRef}}
 
-**`split()`** 方法使用指定的分隔符字符串将一个{{jsxref("String")}}对象分割成子字符串数组，以一个指定的分割字串来决定每个拆分的位置。
+**`split()`** 方法接受一个模式，通过搜索模式将 {{jsxref("String")}} 分割成一个有序的子串列表，将这些子串放入一个数组，并返回该数组。
 
-{{EmbedInteractiveExample("pages/js/string-split.html")}}
+{{EmbedInteractiveExample("pages/js/string-split.html", "taller")}}
 
 ## 语法
 
-```js
-str.split([separator[, limit]])
+```js-nolint
+split(separator)
+split(separator, limit)
 ```
 
-> **警告：** 如果使用空字符串 (“) 作为分隔符，则字符串不是在每个用户感知的字符 (图形素集群) 之间，也不是在每个 Unicode 字符 (代码点) 之间，而是在每个 UTF-16 代码单元之间。这会摧毁代理对。还请参见[how do you get a string to a character array in javascript](https://stackoverflow.com/questions/4547609/how-do-you-get-a-string-to-a-character-array-in-javascript/34717402#34717402)
-
 ### 参数
 
 - `separator`
-  - : 指定表示每个拆分应发生的点的字符串。`separator` 可以是一个字符串或[正则表达式](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/RegExp)。如果纯文本分隔符包含多个字符，则必须找到整个字符串来表示分割点。如果在 str 中省略或不出现分隔符，则返回的数组包含一个由整个字符串组成的元素。如果分隔符为空字符串，则将 str 原字符串中每个字符的数组形式返回。
-- `limit`
-  - : 一个整数，限定返回的分割片段数量。当提供此参数时，split 方法会在指定分隔符的每次出现时分割该字符串，但在限制条目已放入数组时停止。如果在达到指定限制之前达到字符串的末尾，它可能仍然包含少于限制的条目。新数组中不返回剩下的文本。
+  - : 描述每个分割应该发生在哪里的模式。可以是 `undefined`，一个字符串，或者一个具有 [`Symbol.split`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol/split) 方法的对象——典型的例子是{{jsxref("Global_Objects/RegExp", "正则表达式", "", 1)}}。省略 `separator` 或传递 `undefined` 会导致 `split()` 返回一个只包含所调用字符串数组。所有不是 `undefined` 的值或具有 `@@split` 方法的对象都被[强制转换为字符串](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#字符串强制转换)。
```

Comment:
See step 2 in https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.split

Let's make the description more accurate.

```suggestion
  - : 描述每个分割应该发生在哪里的模式。可以是 `undefined`，一个字符串，或者一个具有 [`Symbol.split`](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Symbol/split) 方法的对象——典型的例子是{{jsxref("Global_Objects/RegExp", "正则表达式", "", 1)}}。省略 `separator` 或传递 `undefined` 会导致 `split()` 返回一个只包含所调用字符串数组。所有不是 `undefined` 的值或不具有 `@@split` 方法的对象都被[强制转换为字符串](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/String#字符串强制转换)。
```